config/src/main/scala/io/buoyant/config/PolymorphicConfig.scala

```diff
@@ -0,0 +1,21 @@
+package io.buoyant.config
+
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonTypeInfo}
+
+/**
+ * A trait that defines the property "kind" as both a var on the object
+ *   and as type information for JSON de/serialization.
+ *
+ * Config objects that expect to be subclassed by multiple other configs
+ *   with different "kind" values should extend this trait.
+ */
+@JsonTypeInfo(
```

Comment:
I'm actually surprised this works because as far as I can tell, annotations are not inherited.  But it does seem to work and is a great reduction in boilerplate.  Must be some jackson magic going on.  🚀